### PR TITLE
API-22759: Update Benefits Intake Email Report Recipients (Includes Refactoring)

### PR DESCRIPTION
**Pull request by kristen-brown**
## Summary
This PR updates the email report recipients for `VBADocuments` (Benefits Intake) since the API has transitioned teams from Team Caracal to Team Banana Peels. It also includes refactoring for the reporting functions, adds specs around those same functions, and moves a Slack alert email into the app's settings.

## Related issue(s)
[API-22759](https://vajira.max.gov/browse/API-22759)
[Related devops PR](https://github.com/department-of-veterans-affairs/devops/pull/12441)
[Related vsp-infra-application-manifests PR](https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/1264)

## Testing done
After this PR is deployed, the email reporting behavior for `VBADocuments` should remain the same other than sending the emails to an updated recipient list. I made sure the unit testing coverage was improved for the core areas that I was changing. The majority of the behavior can be verified on staging, and the difference in production behavior (specifically, adding a Slack email address to the recipients) is not key functionality and can verified once on production.

## Screenshots
N/A

## What areas of the site does it impact?
This PR impacts the `VBADocuments` (Benefits Intake) module, and specifically the email reporting functions.

## Acceptance criteria
- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation) – N/A
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable) – N/A
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature – N/A

## Requested Feedback
No specific feedback requested for this PR.

### CODE REVIEW

**Review by caseywilliams — APPROVED** [review]
Makes sense, looks good to me!

File: modules/vba_documents/app/mailers/vba_documents/unsuccessful_report_mailer.rb [hunk 1]
```diff
@@ -9,9 +9,17 @@ def self.fetch_recipients
...
+
+      recipients_for_current_env = recipients[env.to_s] || []
+      recipients_for_all_envs = recipients['common'] || []
+
+      if Settings.vba_documents.slack.enabled
+        slack_alert_email = Settings.vba_documents.slack.default_alert_email
+        (recipients_for_current_env + recipients_for_all_envs).uniq.append(slack_alert_email).compact
+      else
+        (recipients_for_current_env + recipients_for_all_envs).uniq
+      end
```
Comment (kristen-brown) [inline modules/vba_documents/app/mailers/vba_documents/unsuccessful_report_mailer.rb:22]: You'll notice the duplicated code here. Next time I touch these files I plan to move the "fetch recipients" functionality to a separate Concern.

File: modules/vba_documents/spec/mailers/monthly_report_mailer_spec.rb [hunk 2]
```diff
@@ -25,4 +26,52 @@
...
+
+    context 'when environment is staging' do
+      let(:expected_result) do
+        %w[
+          road.runner@va.gov
+          tweety.bird@va.gov
+          bugs.bunny@va.gov
+          sylvester.cat@va.gov
+        ]
+      end
```
Comment (cilestin) [inline modules/vba_documents/spec/mailers/monthly_report_mailer_spec.rb:64]: What happens when the env is `dev` which has no recipients for that env?

Comment (kristen-brown) [reply modules/vba_documents/spec/mailers/monthly_report_mailer_spec.rb:64]: Good call, I should add a test for that. `dev` would return the common recipients only.

Comment (kristen-brown) [reply modules/vba_documents/spec/mailers/monthly_report_mailer_spec.rb:64]: These have been added!

**Review by michelpmcdonald — APPROVED** [review]

**Pull request merged**